### PR TITLE
Add Search Applications API

### PR DESCRIPTION
**Pull request by carlosdelest**
Add Search Applications API, including both CRUD and `_search` request.

### CODE REVIEW

**Review by kderusso — COMMENTED** [review]
Looks great, thanks for doing this @carlosdelest ! One comment.

File: output/schema/schema.json [hunk 1]
```diff
@@ -10060,12 +10060,19 @@
...
-      "response": null,
+      "response": {
+        "name": "Response",
+        "namespace": "search_application.delete"
+      },
       "responseMediaType": [
         "application/json"
       ],
+      "since": "8.8.0",
       "stability": "beta",
```
[the remark this answers is not in the document]
Comment (carlosdelest) [reply output/schema/schema.json:10076]: I was thinking on having first input from the clients team - we need to do that change in the ES specs as well so it gets propagated here

[the remark this answers is not in the document]
Comment (kderusso) [reply output/schema/schema.json:10076]: The API docs have already been updated! https://github.com/elastic/elasticsearch/pull/95308

**Review by sethmlarson — COMMENTED** [review]
Looks good to me, only had one comment:

File: specification/search_application/search/SearchApplicationsSearchRequest.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * @stability beta
+ */
+interface Request extends RequestBase {
+  path_parts: {
+    /**
+     * The name of the search application to be searched
+     */
+    name: Name
+  }
+  /** @codegen_name search_params */
```
[the remark this answers is not in the document]
Comment (carlosdelest) [reply specification/search_application/search/SearchApplicationsSearchRequest.ts:37]: Thanks @sethmlarson ! There were validation errors that pointed to using `codegen_name` so I included that. Removing this!

**Comment by carlosdelest** [comment]
Closing this PR to open from the same repo, so Github actions are successful

**Pull request closed without merging**